apps/amo/utils.py

```diff
@@ -770,11 +771,11 @@ def no_translation(lang=None):
     """
     old_lang = translation.trans_real.get_language()
     if lang:
-        translation.trans_real.activate(lang)
+        tower.activate(lang)
```

Comment:
Tower is doing clever things under the hood, so we need to use it to "really" (de)activate a locale.